samples/test/downscoping-with-cab.test.js

```diff
@@ -21,121 +21,36 @@
-          },
-        },
-      ],
-    },
-  };
-
-  it('should only have access to the object specified in the cab rule', async () => {
-    const projectId = await googleAuth.getProjectId();
-    const errorMessage =
-      'does not have storage.objects.get access to the Google Cloud Storage object.';
-
-    // Create the OAuth credentials (the consumer).
-    const oauth2Client = new OAuth2Client();
-    // We are defining a refresh handler instead of a one-time access
-    // token/expiry pair.
-    // This will allow the consumer to obtain new downscoped tokens on
-    // demand every time a token is expired, without any additional code
-    // changes.
-    oauth2Client.refreshHandler = async () => {
-      // Obtain an authenticated client via ADC.
-      const client = await googleAuth.getClient();
-
-      // Create a DownscopedClient with the credential access boundary defined
-      // above. This client will only be able to access the file contents of
-      // the specified object.
-      const cabClient = new DownscopedClient(client, cab);
-
-      // The common pattern of usage is to have a token broker pass the
-      // downscoped short-lived access tokens to a token consumer via some
-      // secure authenticated channel. For illustration purposes, we are
-      // generating the downscoped token locally.
-      const refreshedAccessToken = await cabClient.getAccessToken();
-      return {
-        access_token: refreshedAccessToken.token,
-        expiry_date: refreshedAccessToken.expirationTime,
-      };
-    };
-
-    const storageOptions = {
-      projectId,
-      authClient: {
-        sign: () => {
-          Promise.reject('unsupported');
-        },
-        getCredentials: async () => {
-          Promise.reject();
-        },
-        request: opts => {
-          return oauth2Client.request(opts);
-        },
-        authorizeRequest: async opts => {
-          opts = opts || {};
-          const url = opts.url || opts.uri;
-          const headers = await oauth2Client.getRequestHeaders(url);
-          opts.headers = Object.assign(opts.headers || {}, headers);
-          return opts;
-        },
-      },
-    };
-
-    const storage = new Storage(storageOptions);
-    // Test object1 can be downloaded.
-    const downloadFile = await storage
-      .bucket(bucketName)
-      .file(objectName1)
-      .download();
-    assert.strictEqual(downloadFile.toString('utf8').includes(CONTENT), true);
-
-    // Test object2 download fails due to no access.
-    try {
-      await storage.bucket(bucketName).file(objectName2).download();
-      throw new Error('should only have access to the object 2');
-    } catch (e) {
-      assert.strictEqual(e.message.includes(errorMessage), true);
-    }
+  it('should have access to the object specified in the cab rule', async () => {
+    const output = await execAsync(
+      `${process.execPath} downscopedclient ${bucketName} ${objectName}`
```

Comment:
I don't think the 2 arguments (bucket name and object name) will be plugged in automatically into the main function above (in the sample). There isn't any logic to parse them there.